src/main/java/com/uid2/shared/store/reader/RotatingClientKeyProvider.java

```diff
@@ -55,14 +58,20 @@ public long getVersion(JsonObject metadata) {
 
     @Override
     public long loadContent(JsonObject metadata) throws Exception {
-        return reader.loadContent(metadata, "client_keys");
+        long version = reader.loadContent(metadata, "client_keys");
+        this.authorizableStore.refresh(this.getAll());
```

Comment:
Are we now essentially maintaining two different "snapshots" that are duplicated? `AuthorizableStore` 's records are loaded in by `reader`'s snapshot, would we still need `reader`'s snapshot?

Reply:
ScopedReader maintains its own snapshot of what its parsed, which might be necessary for other users for ScopedReader

It's only the references that are duplicated, I don't see an issue with that. That's the cost of abstraction sometimes

Reply:
I see, I brought up the question because `RotatingOperatorKeyProvider` does it a bit differently without `ScopedReader` - maybe it would be investigating whether it would be good to make this consistent on the operator key side

Reply:
Ah right, I think this scoped reader stuff is for site specific files, I don't think Operators are site specific so they didn't need it 

For Operator I guess you can just pass in a collection of Operators directly, the RotatingOperatorKeyProvider won't need to store a duplicate list of references

Reply:
Makes sense